src/vcpkg/tools.cpp

```diff
@@ -627,16 +627,18 @@ namespace vcpkg
         {
         }
 
-        template<typename Func>
+        template<typename Func, typename Func2>
         Optional<PathAndVersion> find_first_with_sufficient_version(MessageSink& status_sink,
                                                                     const ToolProvider& tool_provider,
                                                                     const std::vector<Path>& candidates,
-                                                                    Func&& accept_version) const
+                                                                    Func&& accept_version,
+                                                                    Func2&& log_candidate) const
```

Comment:
This should not be `&&`, because if it is an rvalue, calling it might 'destroy' it.

Yes I know the other one is already wrong.

It should get a comment describing the call interface.